### PR TITLE
Crayons: buttons followup

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Optimization
- [x] Documentation Update

## Description

This PR improves our recently introduced buttons. This PR does following:
1. It introduces new button variant: `secondary` - which visually is between `default` and `primary`. It is something that Design identified as actually missing while working on few mockups.
    - Therefore I turned `primary` boolean into `variants` select.
2. Icons can now be used not only with block links but also inline links. This limitation was kinda silly...
3. It fixes few things in documentation.

## QA Instructions, Screenshots, Recordings

Spin up storybook `yarn storybook` and click around Buttons, Links and CTAs.

## Added/updated tests?

- [x] Yes, updated

## [Forem core team only] How will this change be communicated?

- [x] Forem.team post
- [x] Storybook update.

### CODE REVIEW

File: app/assets/javascripts/initializers/initializePodcastPlayback.js [hunk 1]
```diff
@@ -475,7 +475,9 @@ function initializePodcastPlayback() {
 
   function handlePodcastMessages(event) {
     const message = JSON.parse(event.detail);
-    if (message.namespace !== 'podcast') { return }
```
Comment (ludwiczakpawel) [inline app/assets/javascripts/initializers/initializePodcastPlayback.js:478]: Some auto-formatting sneaked in...

File: app/javascript/crayons/Buttons/Button.jsx [hunk 2]
```diff
@@ -73,7 +73,7 @@ ButtonNew.displayName = 'ButtonNew';
 
 ButtonNew.propTypes = {
   children: defaultChildrenPropTypes,
-  primary: PropTypes.bool,
+  variant: PropTypes.oneOf([undefined, 'primary', 'secondary']),
```
Comment (nickytonline) [inline app/javascript/crayons/Buttons/Button.jsx:76]: Refactor (blocking): `variant` is optional, so `undefined` is not required in `PropTypes.oneOf(...)`

**Review by nickytonline — CHANGES_REQUESTED** [review]
One change request for now. I'm going to pull this down to check it out.

File: app/javascript/crayons/Buttons/Button.jsx (hunk 2 above)
Comment (ludwiczakpawel) [reply app/javascript/crayons/Buttons/Button.jsx]: Fixed!

File: app/javascript/crayons/Buttons/Button.jsx [hunk 3]
```diff
@@ -30,8 +30,8 @@ export const ButtonNew = (props) => {
   };
 
   const classes = classNames('c-btn', {
-    'c-btn--primary': primary,
-    'c-btn--destructive': destructive,
+    [`c-btn--${variant}`]: variant,
+    'c-btn--destructive': destructive && variant !== 'secondary',
```
Comment (nickytonline) [inline app/javascript/crayons/Buttons/Button.jsx:34]: Should a destructive button be of a secondary variant? Shouldn't it always be a primary button?

Comment (ludwiczakpawel) [reply app/javascript/crayons/Buttons/Button.jsx:34]: No, we already have a case where we use this lighter style for destructive button (see "Delete" next to drafts posts in Dashboard) and it does seem like we gonna have few more later on, in Admin side.

File: app/javascript/crayons/Buttons/__stories__/Buttons.mdx [hunk 4]
```diff
@@ -14,35 +14,50 @@ That is: `<CTA />` or `<Link />` should be used where navigation is elicited, an
...
 However, 'Register' call-to-action—taking the user to a different form, in a different page or screen—would benefit from `<CTA />` or `<Link />` semantics.
 
-Do not use `<Button />s` as navigational elements. Instead, use Links (or CTAs) when the desired action is to take the user to a new page.
+Do not use `<Button />s` as navigational elements. Instead, use `<Link />s` (or `<CTA />s`) when the desired action is to take the user to a new page.
 
 ## Variants
 
+### Default
+
+_Default_ variant is not very prominent, and therefore can be used for less prominent actions.
```
Comment (nickytonline) [inline app/javascript/crayons/Buttons/__stories__/Buttons.mdx:23]: Wording (blocking): Incomplete sentence. Start with `The`. There are other instances of this below as well in the rest of the documentation.

```suggestion
The _Default_ variant is not prominent, and therefore can be used for less prominent actions.
```

Wording (nitpick):

```suggestion
The _Default_ variant is not prominent. Use it potentially for less noticeable actions.
```

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: app/javascript/crayons/Buttons/__stories__/Buttons.mdx (hunk 4 above)
Comment (ludwiczakpawel) [reply app/javascript/crayons/Buttons/__stories__/Buttons.mdx]: _The_ vs _A_ was always my weak point in English... :( I tried to update it few examples, but... 🤷‍♂️

**Review by aitchiss — APPROVED** [review]
Looks good! Just one suggestion, but happy to approve 🚀

File: app/javascript/crayons/Buttons/Button.jsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ import { Icon } from '@crayons';
 export const ButtonNew = (props) => {
   const {
     children,
-    primary,
+    variant,
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/javascript/crayons/Buttons/Button.jsx:11]: good suggestion, updating.

**Review by fdoxyz — APPROVED** [review]
These look great on storybook and the explanations seem very helpful! ✨ 

I have no opinions on the implementation per-se though. LGTM!

File: app/javascript/crayons/Buttons/__stories__/Buttons.mdx [hunk 6]
```diff
@@ -14,35 +14,50 @@ That is: `<CTA />` or `<Link />` should be used where navigation is elicited, an
...
+The _Default_ buttons are visually light (greyish) and you can have as many of them as you want (within common sense :D). 
+They can (but don't have to) be used in conjunction with a _Primary_ button and less often with _Secondary_ buttons (use your common sense again or ping your designer).
+When combined with the _Primary_ or the _Secondary_, the _Default_ variant should be used for any remaining actions.
+
+Whenever used, make sure it has sufficient color contrast against the background it's used on.
+
 ### Primary
 
-Primary button has special look: high contrast (against rest of the view/component) and it's filled with accent color. 
+The _Primary_ variant has special look: high contrast (against rest of the view/component) and it's filled with accent color.
```
Comment (juliannatetreault) [inline app/javascript/crayons/Buttons/__stories__/Buttons.mdx:32]: **_Quick suggestion_**: there's an "a" missing here. 😊 
```suggestion
The _Primary_ variant has a special look: high contrast (against rest of the view/component) and it's filled with accent color. 
```

File: app/javascript/crayons/Buttons/__stories__/Buttons.mdx [hunk 7]
```diff
@@ -14,35 +14,50 @@ That is: `<CTA />` or `<Link />` should be used where navigation is elicited, an
...
 
-Ideally there should be only one Primary button per entire component, or sometimes even per entire view.
-Primary buttons should be used for the principal call to action on the page.
+Ideally there should be only one _Primary_ button per entire component, or sometimes even per entire view.
+Primary buttons should be used for the principal action on the page.
 
-### Default
+### Secondary
+
+The _Secondary_ variant is visually in between _Primary_ and _Default_ buttons. It's not as loud as _Primary_ button but definitely stands out more than _Default_ button.
```
Comment (juliannatetreault) [inline app/javascript/crayons/Buttons/__stories__/Buttons.mdx:39]: **_Quick suggestion_**: let's add "the" before the button names here for consistency and readability. 😊 
```suggestion
The _Secondary_ variant is visually in between the _Primary_ and the _Default_ buttons. It's not as loud as the _Primary_ button but definitely stands out more than the _Default_ button.
```

File: app/javascript/crayons/Buttons/__stories__/Buttons.mdx [hunk 8]
```diff
@@ -14,35 +14,50 @@ That is: `<CTA />` or `<Link />` should be used where navigation is elicited, an
...
 ### Destructive actions
 
 For actions that could have destructive effects on the user’s data (for example, *delete* or *remove*) we have a special style.
-This style can be applied by using `destructive` prop and it will work for both: primary and default variants. Take care when using the default variant to ensure that the red text has sufficient color contrast against the background where you are using it.
+This style can be applied by using `destructive` prop and it will work with two variants: _Primary_ and _Default_ (as it doesn't really make sense to have _Secondary_ variant for destructive buttons).
 
 ### Icons
 
-Icon can be placed in front of a label to both clarify an action and call attention to a button. 
+Icon can be placed in front of a label to both clarify an action but also to make button stand out more.
```
Comment (juliannatetreault) [inline app/javascript/crayons/Buttons/__stories__/Buttons.mdx:58]: Quick suggestion: let's make "icon" plural here and add the word "the"!
```suggestion
Icons can be placed in front of a label to both clarify an action but also to make the button stand out more.
```

File: app/javascript/crayons/CTAs/__stories__/CTAs.mdx [hunk 9]
```diff
@@ -16,12 +16,11 @@ Do not use `<Button />s` as navigational elements. Instead, use Links (or CTAs)
 A CTA renders an anchor element, therefore standard link markup must be used,
 including a `href` attribute, otherwise, it is not focusable by the keyboard.
 
-
 ### Icons
 
-Icon can be placed in front of a label to both clarify an action and call attention to a CTA.
+Icon can be placed in front of a label to both clarify an action but also to make CTA stand out more.
```
Comment (juliannatetreault) [inline app/javascript/crayons/CTAs/__stories__/CTAs.mdx:21]: Quick suggestion: let's make "Icon" plural here, too!
```suggestion
Icons can be placed in front of a label to both clarify an action but also to make a CTA stand out more.
```

File: app/javascript/crayons/Links/__stories__/Links.mdx [hunk 10]
```diff
@@ -27,10 +27,9 @@ It is recommended to always use `branded` variant for inline links, especially w
 
 ### Icons
 
-Icons are only supported in *block* links (`block`).
-Icon can be placed in front of a label to both clarify an action and call attention to a link.
-By passing an object in `src` prop, there will be automatically included `<Icon />` component.
-By default, the icon itself won't be customizable.
+Icon can be placed in front of a label to both clarify an action but also to make link stand out more.
```
Comment (juliannatetreault) [inline app/javascript/crayons/Links/__stories__/Links.mdx:30]: **_Quick suggestion_**: (same as my last two suggestions!) let's make "Icon" plural here and add an "a" before "link" (or remove the "a" and make "link" plural instead 😄 )!
```suggestion
Icons can be placed in front of a label to both clarify an action but also to make a link stand out more.
```

**Review by juliannatetreault — COMMENTED** [review]
Left a few quick suggestions, but otherwise, this is super informative and looks great! ✨

**Review by juliannatetreault — APPROVED** [review]

File: app/javascript/crayons/Buttons/__stories__/Buttons.mdx (hunk 8 above)
Comment (ludwiczakpawel) [reply app/javascript/crayons/Buttons/__stories__/Buttons.mdx:58]: But you can only have one icon in a button?

**Review by nickytonline — APPROVED** [review]
Looks great @ludwiczakpawel! 🚀

**Pull request merged**